bionic/task_state.py

```diff
@@ -339,21 +336,33 @@ def initialize(self, bootstrap, flow_instance_uuid):
             for task_key in self.task_keys
         ]
 
+        can_persist = (
+            single_unique_element(
+                entity_def.can_persist
+                for entity_def in self.entity_defs_by_dnode.values()
+            )
+            and not self.output_would_be_missing()
+        )
+        if can_persist and bootstrap is None:
+            descriptors = [
+                task_key.dnode.to_descriptor() for task_key in self.task_keys
+            ]
+            message = f"""
+            Descriptors {descriptors!r} are set to be persisted but that's
+            impossible; the cache is not available yet because core
```

Comment:
English nit: this semicolon probably makes more sense as a colon, because the second part of the sentence explains the first.